File: chapter10/src/main/java/com/example/chapter10/PageFragment.java
package com.example.chapter10;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.ViewStub;

/**
 * @author wangzhichao
 * @date 2019/09/19
 */
public class PageFragment extends Fragment {
    private static final String ARGS_LAYOUT_RES = "args_layout_res";

    public static PageFragment newInstance(int layoutRes) {

        Bundle args = new Bundle();
        args.putInt(ARGS_LAYOUT_RES, layoutRes);
        PageFragment fragment = new PageFragment();
        fragment.setArguments(args);
        return fragment;
    }
    private int layoutRes;
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            layoutRes = getArguments().getInt(ARGS_LAYOUT_RES);
        }
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_page, container, false);
        ViewStub viewStub = view.findViewById(R.id.viewStub);
        viewStub.setLayoutResource(layoutRes);
        viewStub.inflate();
        return view;
    }
}

File: chapter02/src/main/java/com/example/chapter02/PageFragment.kt
package com.example.chapter02

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.ViewStub
import androidx.fragment.app.Fragment
import androidx.viewpager.widget.ViewPager
import com.example.chapter02.part5_frame_animation.FrameAnimXmlViewGroup
import com.example.chapter02.part5_frame_animation.FrameAnimCodeViewGroup
import com.example.chapter02.part2_view_animation_code.ViewAnimationCodeControlListenerViewGroup

/**
 * @author wangzhichao
 * @date 2019/09/17
 */
class PageFragment : Fragment() {
    private var layoutRes = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            layoutRes = arguments!!.getInt(ARGS_LAYOUT_RES)
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_page, container, false)
        val viewStub = view.findViewById<ViewStub>(R.id.viewStub)
        viewStub.layoutResource = layoutRes
        viewStub.inflate()
        return view
    }

    companion object {
        private const val ARGS_LAYOUT_RES = "args_layout_res"
        fun newInstance(layoutRes: Int): PageFragment {
            val args = Bundle()
            args.putInt(ARGS_LAYOUT_RES, layoutRes)
            val fragment = PageFragment()
            fragment.arguments = args
            return fragment
        }
    }
}